src/prefect/config.toml

```diff
@@ -52,7 +52,7 @@ eager_edge_validation = false
 # If false, only a single execution will occur (no retries, etc.)
 run_on_schedule = true
 # If true, tasks which set `checkpoint=True` will have their result handlers called
-checkpointing = false
+checkpointing = true
```

Comment:
I think we should keep this `False` for now; this will be a large breaking change for anyone using non-JSON compatible `Parameter` values with Core.